ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs

```diff
@@ -79,9 +83,21 @@ type EraCrypto era = Crypto era
 -- needed to determine the hard fork point. In the future this should be
 -- replaced with an appropriate API - see
 -- https://github.com/input-output-hk/ouroboros-network/issues/2890
+--
+-- TODO Currently we include the constraint @SL.AdditionalGenesisConfig era ~
+-- ()@. When we fork to Alonzo we will need additional genesis config
+-- information.
 class ( SL.ShelleyBasedEra era
       , State (LC.EraRule "PPUP" era) ~ SL.PPUPState era
       , Default (State (LC.EraRule "PPUP" era))
+      , HasField "_maxBHSize" (LC.PParams era) Natural
+      , HasField "_maxTxSize" (LC.PParams era) Natural
+      , HasField "_a0" (LC.PParams era) Rational
+      , HasField "_nOpt" (LC.PParams era) Natural
+      , HasField "_rho" (LC.PParams era) SL.UnitInterval
+      , HasField "_tau" (LC.PParams era) SL.UnitInterval
+      , HasField "_protocolVersion" (SL.PParamsDelta era) (SL.StrictMaybe SL.ProtVer)
+      , SL.AdditionalGenesisConfig era ~ ()
```

Comment:
What do you think about adding this field now instead of adding this temporary constraint? (Plus alignment...)

The _existing_ occurrences of the `ProtocolParamsShelleyBased` ctor would have to pass `()`, while the eventual one for `AlonzoEra` would be non-trivial.

```
--- a/ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs
+++ b/ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs
@@ -225,6 +225,7 @@ data ProtocolParamsShelleyBased era = ProtocolParamsShelleyBased {
       -- mutually incompatible.
     , shelleyBasedInitialNonce      :: SL.Nonce
     , shelleyBasedLeaderCredentials :: [TPraosLeaderCredentials (EraCrypto era)]
+    , shelleyBasedAdditionalGenesisConfig :: SL.AdditionalGenesisConfig era
     }
```

Reply:
Yeah, that seems reasonable to me, I added the temporary constraint because I wasn't sure how you would want to handle this.

Reply:
One more thing on top of this (knowing this is a bit of bikeshedding), could ledger-spec introduce a data type 

```
data NoAddidtionalConfig = NoAdditionalConfig
```

This is equivalent to `()` but comes with a nicer readability on the call site. We can then say things like

```
SL.AdditionalGenesisConfig era ~ SL.NoAdditionalConfig
```

or when there will be functions that take `SL.AdditionalGenesisConfig`  instead of passing unit to them (for eras without additional config) we would pass a nice `NoAddidtionalConfig`.

Syntactic change, without changing the semantics.

Reply:
We agreed to do this in a follow-up tech debt PR.